man/systemd.timer.xml

```diff
@@ -288,6 +288,21 @@
         multiple times. Defaults to
         <varname>yes</varname>.</para></listitem>
       </varlistentry>
+
+      <varlistentry>
+        <term><varname>PersistentRandomizedDelay=</varname></term>
+
+        <listitem><para>Takes a boolean argument. If true, when the timer
+        is activated, the service unit is not triggered immediately even if
+        it would have been triggered at least once during the time when the
+        timer was inactive. Instead, the time when the timer was activated
+        is set as the next determined elapsing time, and then a randomly
+        selected delay by RandomizedDelaySec= will be add to that time. Note
+        that this setting only has an effect on timers configured with
+        <varname>Persistent=</varname> and <varname>RandomizedDelaySec=</varname>.
+        Defaults to <varname>false</varname>.
```

Comment:
I'm not sure I understand... the timer would be triggered "at the next elapsing time" so we do miss the trigger that would have happened when the machine was off ?

Wouldn't it make more sense to trigger it "RandomizedDelaySec" after ther restart instead of waiting for the next trigger ?

Alternatively, a PersistantRandomizedDelaySec= which defaults to 0 could be a good approch too...

Reply:
I'm sorry, my English is very bad...

> Wouldn't it make more sense to trigger it "RandomizedDelaySec" after ther restart instead of waiting for the next trigger ?

Yes! This should work exactly like so.
e.g.

| date | time | state |
| --- | --- | --- |
| Yesterday | 22:00:00 | Poweroff |
| Today | 00:03:25 | Original scheduled time (+RandomizedDelaySec) but machine was off. |
| Today | 07:00:00 | Boot. If PersistentRandomizedDelaySec=false, the service will triggered immidietly. If PersistentRandomizedDelaySec=true, the service will not triggered yet here. Insted, "now" +RandomizedDelaySec will be the new scheduled time. |
| Today | 07:06:18 | Service will triggered! |

Reply:
awesome, in that case I think you need to reword a bit the documentation, since I understood exactly the opposite :)

how about:

When both Persistant= and RandomizedDelaySec are set and a trigger is missed because the machine is powered off, this variable decide whether the timer will be triggered right away on startup or if RandomizedDelaySec will be applied before triggering the missed timer.

(I still think a separate PersistantRandomizedDelaySec would be better, but it's not critical)

Reply:
Thank you, I'll rewrite it!

Two options, RandomizedDelaySec and PersistentRandomizedDelaySec, seem slightly redundant for me...